unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendsHUD/Scripts/FriendsHUDView.cs

```diff
@@ -31,14 +65,106 @@ public static FriendsHUDView Create(FriendsHUDController controller)
         return view;
     }
 
-    internal List<FriendEntryBase> GetAllEntries()
+    public List<FriendEntryBase> GetAllEntries()
     {
         var result = new List<FriendEntryBase>();
         result.AddRange(friendsList.GetAllEntries());
         result.AddRange(friendRequestsList.GetAllEntries());
         return result;
     }
 
+    public FriendEntryBase GetEntry(string userId)
+    {
+        return friendsList.GetEntry(userId) ?? friendRequestsList.GetEntry(userId);
+    }
+
+    public void UpdateEntry(string userId, FriendEntryBase.Model model)
+    {
+        friendsList.UpdateEntry(userId, model);
+        friendRequestsList.UpdateEntry(userId, model);
+    }
+
+    public void DisplayFriendUserNotFound() => friendRequestsList.DisplayFriendUserNotFound();
+
+    public bool IsFriendListFocused() => friendsList.gameObject.activeInHierarchy;
+
+    public int GetReceivedFriendRequestCount() => friendRequestsList.receivedRequestsList.Count();
+
+    public void Destroy()
+    {
+        if (!this) return;
+        if (!gameObject) return;
+        Destroy(gameObject);
+    }
+
+    public void Show()
+    {
+        gameObject.SetActive(true);
+        
+        if (friendsButton.interactable)
+            friendsButton.onClick.Invoke();
+    }
+
+    public void Hide() => gameObject.SetActive(false);
+    
+    public void UpdateFriendshipStatus(string userId, FriendshipAction friendshipAction,
+        FriendEntryBase.Model friendEntryModel)
+    {
+        switch (friendshipAction)
```

Comment:
Could it be worth having a default case which throws an error to simplify future cases in which new actions could be added?

Reply:
i will add an error log so it does not go unnoticed.. throwing an exception may break the flow if not handled properly